gradio/blocks.py

```diff
@@ -720,6 +720,11 @@ def __init__(
             if analytics_enabled is not None
             else analytics.analytics_enabled()
         )
+        if wasm_utils.IS_WASM and self.analytics_enabled:
+            self.analytics_enabled = False
+            warnings.warn(
+                "Analytics are not supported in the Wasm mode."
+            )
```

Comment:
```suggestion
        if wasm_utils.IS_WASM:
            self.analytics_enabled = False
```

Reply:
That being said, it *would* be nice to know how much usage `gradio-wasm` is getting. We can get downloads from npm right?

Reply:
That may be right, but the reason I disabled the analytics now is that the analytics feature does not work on Pyodide because it uses the `threading` module.

> We can get downloads from npm right?

Yes. Also, the CDN jsDelivr should also provides such stats, while its page is not found somehow (I don't know why... maybe because the downloads are still few? For exmaple, my `@stlite/mountable` has its [jsDelivr page](https://www.jsdelivr.com/package/npm/@stlite/mountable) and we can see the stats).

* NPM: https://www.npmjs.com/package/@gradio/lite
* jsDeliver: (currently available now)

Reply:
So, I thought disabling the analytics is the simplest solution to make `gradio-lite` work,
but fixing the `analytics` module to be Wasm-compatible is another way if we think the analytics is important on `gradio-lite` too.

It would be fixing the code depending on the`threading` module to depend on `asyncio` instead, and replacing the `requests` module with `pyodide.http.pyfetch`.

Reply:
> but fixing the analytics module to be Wasm-compatible is another way if we think the analytics is important on gradio-lite too.

> It would be fixing the code depending on thethreading module to depend on asyncio instead, and replacing the requests module with pyodide.http.pyfetch.

This would be great! I do think analytics capturing the usage would be very valuable. We don't have to add them in this PR, but would be good to have before launch

Reply:
I created another PR doing it: #5045

However let me ask one question:
While it is a higher-level discussion, should we enable the analytics feature on the Wasm ver too?
My concern is, though not 100% sure, users may expect the Wasm ver. is similar to being "standalone" and such external network access does not occur.

Reply:
Also, is it legally clear (sorry I'm not a legal expert though)?
In the Wasm mode, the analytics telemetry including IP address is sent from users' devices, and it may conflict with something like GDPR (and Hugging Face has a legal entity in EU)?

Reply:
Good question @whitphx I think its important to have analytics to get an idea to have usage of the wasm mode. We have an existing issue to make sure analytics is compliant with GDPR (#4226) but for now, any user can disable analytics pretty easily so I think it should be fine.